src/metabase/models/user.clj

```diff
@@ -102,10 +102,13 @@
     email       (update :email u/lower-case-en)))
 
 (defn add-common-name
-  "Add a `:common_name` key to `user` by combining their first and last names."
-  [{:keys [first_name last_name], :as user}]
-  (cond-> user
-    (or first_name last_name) (assoc :common_name (str first_name " " last_name))))
+  "Add a `:common_name` key to `user` by combining their first and last names, or using their email if names are `nil`."
+  [{:keys [first_name last_name email], :as user}]
+  (let [common-name (if (or first_name last_name)
+                      (str/trim (str first_name " " last_name))
+                      email)]
+    (merge user
+           (when common-name {:common_name common-name}))))
```

Comment:
The common name is used by the frontend sometimes, and needs to fall back to a user's email if there aren't first or last names.

Reply:
It's not possible for `common-name` to be null right? `email` will always exist and `(str/trim ...)` will return a string so `common-name` will always be truthy.

Reply:
It can be nil in a situation like this: `(db/select [User :id])`. The post-select function won't have any of the keys needed to actually assemble the common_name in this case. So, I just wrapped it in a `when` so that it doesn't add `:common_name nil` all the time.

Reply:
Specifically, If you have some user with `{:first_name nil :last_name nil :email "some@email.com"}`, and do `(db/select [User :first_name :last_name])` -> this will default to using email as the common name, but the `:email` key isn't there